src/freenet/support/Fields.java

```diff
@@ -288,23 +288,40 @@ public static long dateTime(String date)
 				deltaType = Calendar.DAY_OF_YEAR;
 			else {
 				String deltaTypeString = date.substring(chop).toLowerCase();
-				if(deltaTypeString.equals("y") || deltaTypeString.equals("year"))
-					deltaType = Calendar.YEAR;
-				else if(deltaTypeString.equals("month") || deltaTypeString.equals("mo"))
-					deltaType = Calendar.MONTH;
-				else if(deltaTypeString.equals("week") || deltaTypeString.equals("w"))
-					deltaType = Calendar.WEEK_OF_YEAR;
-				else if(deltaTypeString.equals("day") || deltaTypeString.equals("d"))
-					deltaType = Calendar.DAY_OF_YEAR;
-				else if(deltaTypeString.equals("hour") || deltaTypeString.equals("h"))
-					deltaType = Calendar.HOUR;
-				else if(deltaTypeString.equals("minute") || deltaTypeString.equals("min"))
-					deltaType = Calendar.MINUTE;
-				else if(deltaTypeString.equals("second") || deltaTypeString.equals("s") || deltaTypeString.equals("sec"))
-					deltaType = Calendar.SECOND;
-				else
-					throw new NumberFormatException(
-						"unknown time/date delta type: " + deltaTypeString);
+                switch (deltaTypeString) {
+                    case "y":
+                    case "year":
+                        deltaType = Calendar.YEAR;
+                        break;
+                    case "month":
+                    case "mo":
+                        deltaType = Calendar.MONTH;
+                        break;
+                    case "week":
+                    case "w":
+                        deltaType = Calendar.WEEK_OF_YEAR;
+                        break;
+                    case "day":
+                    case "d":
+                        deltaType = Calendar.DAY_OF_YEAR;
+                        break;
+                    case "hour":
+                    case "h":
+                        deltaType = Calendar.HOUR;
+                        break;
+                    case "minute":
+                    case "min":
+                        deltaType = Calendar.MINUTE;
+                        break;
+                    case "second":
+                    case "s":
+                    case "sec":
+                        deltaType = Calendar.SECOND;
+                        break;
+                    default:
+                        throw new NumberFormatException(
+                                "unknown time/date delta type: " + deltaTypeString);
+                }
```

Comment:
is the string switch already available in Java 8?

Reply:
Yes, with constant expressions for the case statements. Had to check myself because I also remember something about it being loudly introduced at a much later version but that’s probably for non-constant expressions in the case statements. 🙂